1:1 port from Cloud to Enterprise on Permissions doc (#193)
astronomer/docs: enterprise/next/06_manage-astronomer/03_workspace-permissions.md

The guidelines below will cover:

1. How to invite users to an Astronomer Workspace and Deployment
2. How to view, set and modify user roles
3. Deployment and Workspace Permissions Reference

## Invite Users

Workspace and Deployment _Admins_ can invite and otherwise manage users both via the Astronomer UI and CLI. All users who have access to a Workspace must be assigned 1 of 3 Workspace roles, though deployment-level roles are not required.

Read below for guidelines.

### Invite to Workspace

The ability to invite users to an Astronomer Workspace is limited to Workspace _Admins_, who can also grant the _Admin_ role to other users. Workspace _Editors_ and _Viewers_ cannot invite or otherwise manage other Workspace users, though they may do so at the deployment level depending on their deployment-level role.

A user who creates a Workspace is automatically granted the _Admin_ role for the Workspace and has the ability to create any number of Airflow Deployments within it. Every Workspace must have at least 1 Workspace _Admin_.

#### via Astronomer UI

> To invite a user to a Workspace via [the Astronomer UI](https://app.gcp0001.us-east4.astronomer.io/), navigate to **Workspace** > **Users** > **Invite User**.

Reviewer: ```suggestion
To invite a user to a Workspace via the Astronomer UI, navigate to **Workspace** > **Users** > **Invite User**.
```

URL here is for Cloud-only!

When a Workspace _Admin_ invites a user to a Workspace in which one or more Airflow Deployments exist, they'll have the opportunity to set that user's deployment-level roles as well, though it is not required.

![Invite Workspace User](https://assets2.astronomer.io/main/docs/astronomer-ui/invite-user-workspace.gif)

If a Workspace Admin invites a user to a Workspace that does _not_ have any Airflow Deployments within it, the **Deployment Roles** modal above will not appear.

#### via Astronomer CLI

To invite a user to a Workspace via the Astronomer CLI, run:

```bash
$ astro workspace user add <email-address> --workspace-id=<workspace-id> --role=<workspace-role>
```

Only Workspace _Admins_ can invite other users and set their permissions.

To find **Workspace ID**, you can:

- Run `$ astro workspace list`

> - Find it in the Workspace URL from your browser after the `/w/` (e.g. `https://app.gcp0001.us-east4.astronomer.io/w/<workspace-id>`)

Reviewer: ```suggestion
- Find it in the Workspace URL from your browser after the `/w/` (e.g. `https://app.basedomain/w/<workspace-id>`)
```

#### via Astronomer UI

To invite a Workspace user to an Airflow Deployment via [the Astronomer UI](https://app.gcp0001.us-east4.astronomer.io/), navigate to: **Workspace** > **Deployment** > **Access**.

From there:

1. Type the Workspace user's name in the search bar on top (or click **Show All** to view all users)
2. Select a deployment role from the drop-down menu to the right of the selected user
3. Click the `+` symbol

![Invite Deployment User](https://assets2.astronomer.io/main/docs/astronomer-ui/invite-user-deployment.gif)

#### via Astronomer CLI

To invite a Workspace user to an Airflow Deployment via the Astronomer CLI, run:

```
$ astro deployment user add <email-address> --deployment-id=<deployment-id> --role=<deployment-role>
```

Only Deployment _Admins_ can invite other users and set their permissions.

To find **Deployment ID**, you can:

- Run `$ astro deployment list`

To set a **Role**, add a flag in the following format:

- `--DEPLOYMENT_EDITOR`
- `--DEPLOYMENT_VIEWER`
- `--DEPLOYMENT_ADMIN`

If you do _not_ specify a role in this command, `DEPLOYMENT_VIEWER` will be set by default.

## View and Edit User Roles

### Workspace

#### View Workspace Users

> To view roles within a Workspace via [the Astronomer UI](https://app.gcp0001.us-east4.astronomer.io/), navigate to **Workspace** > **Users**. All Workspace users have access to this view and can see the roles of other users.

Reviewer: ```suggestion
To view roles within a Workspace via the Astronomer UI, navigate to **Workspace** > **Users**. All Workspace users have access to this view and can see the roles of other users.
```

To edit a user's role via the Astro CLI, run:

```bash
$ astro deployment user update <email> --deployment-id=<deployment-id> --role=<deployment-role>
```

> **Note:** A deployment-level role cannot be edited while a Workspace invitation to that user is pending. If you invite a user to a Workpace, you will not be able to modify their permissions until they accept the Workspace invite.

#### Remove Deployment User

To delete a user from an Airflow Deployment via the Astronomer UI, Deployment _Admins_ can click on the red "wastebasket" icon within the **Access** tab shown in the screenshot above.

To delete a user from an Airflow Deployment via the Astro CLI, run:

```bash
$ astro deployment user delete <email> --deployment-id=<deployment-id>
```

## User Permissions Reference

### Workspace

#### Workspace Admin

Workspace _Admins_ are the highest-tiered role at the Workspace level. Admins:

- Can manage users and their permissions in a Workspace
- Can perform CRUD (create, read, update, delete) operations on the Workspace (e.g. delete the Workspace, change its name)
- Can create Airflow Deployments in the Workspace

> - Can manage **Billing**

Reviewer: ```suggestion
```

Reviewer: (Billing not applicable for Enterprise!)

- Can perform CRUD operations on any Service Account in the Workspace

Workspace _Admins_ do not automatically have CRUD access to all Airflow Deployments within it - they must either create or be added to those Deployments and will be held to the restrictions of their deployment-level role. For example, a Workspace _Admin_ could be a Deployment _Viewer_ and not have access to push code to that Deployment.

Every Workspace must have at least 1 Workspace _Admin_.

#### Workspace Editor

Below a Workspace _Admin_, an _Editor_:

- Can access and make changes to the Workspace in the **Settings** tab
- Can perform CRUD operations on any Service Account in the Workspace
- Can create Airflow Deployments in the Workspace
- Cannot manage other users in the Workspace

> - Cannot add, edit or remove information in **Billing**

Reviewer: ```suggestion
```

![No Admin Tab](https://assets2.astronomer.io/main/docs/astronomer-ui/editor_view.png)

#### Deployment Viewer

Deployment _Viewers_ are limited to read-only mode. They:

- Can view Deployment users
- Can view the **Metrics** and **Logs** tab of the Astro UI
- Can access the Airflow UI
- Cannot deploy to, modify, or delete anything within an Airflow Deployment
- Cannot view Airflow task logs

Viewers _cannot_ push code to an Airflow Deployment or create Service Accounts to do so. Attempts to view logs, trigger DAGs, etc. will result in a `403` and an `Access is Denied` message.

> ![Access Denied](https://assets2.astronomer.io/main/docs/astronomer-ui/access_denied.png)

Reviewer: ```suggestion
![Access Denied](https://assets2.astronomer.io/main/docs/astronomer-ui/access_denied.png)

## What's Next

As an Astronomer Enterprise user, you're free to customize all user permissions at the platform-level. For more information, read:

- [Manage Users on Astronomer Enterprise](https://www.astronomer.io/docs/enterprise/stable/manage-astronomer/manage-platform-users#customize-permissions)
- [Integrate an Auth System](https://www.astronomer.io/docs/enterprise/stable/manage-astronomer/integrate-auth-system)
```
